lib/overcommit/hook/pre_commit/line_endings.rb

```diff
@@ -0,0 +1,51 @@
+module Overcommit::Hook::PreCommit
+  # Checks for line endings in files.
+  class LineEndings < Base
+    def run
+      messages = []
+
+      offending_files.map do |file_name|
+        file = File.open(file_name)
+        file.each_line do |line|
+          # remove configured line-ending
+          line.gsub!(/#{config['eol']}/, '')
```

Comment:
This will match all occurrences of `config['eol']` in the line, not just the last occurrence, no?